File: src/LC2025/LC011.java
package LC2025;

public class LC011 {

    public int maxArea(int[] height) {
        int length = height.length;
        int l = 0;
        int r = length - 1;
        int max = 0;
        while (l < r) {
            int lv = height[l];
            int rv = height[r];
            max = Math.max(max, Math.min(lv, rv) * (r - l));
            if (lv < rv) {
                l ++;
            } else {
                r --;
            }
        }
        return max;
    }
}

File: srcKotlin/LCKotlin/LC011.kt
package LCKotlin

class LC011 {

    fun maxArea(height: IntArray): Int {
        var maxRes: Int = 0
        var left: Int = 0
        var right: Int = height.lastIndex

        while (left < right) {
            val leftV = height[left]
            val rightV = height[right]
            maxRes = maxRes.coerceAtLeast(leftV.coerceAtMost(rightV) * (right - left))
            if (leftV <= rightV) {
                left++
            } else {
                right--
            }
        }
        return maxRes
    }
}